osh/word_eval.py

```diff
@@ -902,7 +904,11 @@ def _EvalVarRef(self, val, blame_tok, quoted, vsub_state, vtest_place):
             else:
                 raise error.TypeErr(val, 'Var Ref op expected Str', blame_tok)
 
-        bvs_part = self.unsafe_arith.ParseVarRef(var_ref_str, blame_tok)
+        try:
+            bvs_part = self.unsafe_arith.ParseVarRef(var_ref_str, blame_tok)
+        except error.FatalRuntime as e:
+            raise error.VarSubFailure(e.msg, e.location)
```

Comment:
Note: Since the fatal error (`e_die`) is hardcoded inside `unsafe_arith.ParseVarRef` (`osh/sh_expr_eval.py`), we try to cancel it here.